### ISSUE TITLE
[Power] Segmentation fault in lapack/openblas

### ISSUE THREAD

**Issue by vchuravy** [JuliaLang/julia#35493]
```
LinearAlgebra/lapack               (4) |        started at 2020-04-15T19:43:02.484
      From worker 4:	
      From worker 4:	signal (11): Segmentation fault
      From worker 4:	in expression starting at /buildworker/worker/tester_linuxppc64le/build/share/julia/stdlib/v1.5/LinearAlgebra/test/lapack.jl:197
      From worker 4:	idamax_k at /buildworker/worker/tester_linuxppc64le/build/bin/../lib/julia/libopenblas64_.so (unknown line)
      From worker 4:	idamax_64_ at /buildworker/worker/tester_linuxppc64le/build/bin/../lib/julia/libopenblas64_.so (unknown line)
      From worker 4:	dgebal_64_ at /buildworker/worker/tester_linuxppc64le/build/bin/../lib/julia/libopenblas64_.so (unknown line)
      From worker 4:	gebal! at /buildworker/worker/package_linuxppc64le/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/lapack.jl:209
      From worker 4:	unknown function (ip: 0x3fffa123b45b)
      From worker 4:	_jl_invoke at /buildworker/worker/package_linuxppc64le/build/src/gf.c:2235 [inlined]
      From worker 4:	jl_apply_generic at /buildworker/worker/package_linuxppc64le/build/src/gf.c:2402
      From worker 4:	Allocations: 509405995 (Pool: 509241626; Big: 164369); GC: 481
```

**Comment by vchuravy** [comment]
Alternative reproducer which either seqfaults in the same place or throws a peculiar error:

```
D = Diagonal([[1 2; 3 4], [1 2; 3 4]])
exp(D)
```

```
julia> exp(D)
ERROR: LAPACKException(140736848912640)
Stacktrace:
  [1] chklapackerror(ret::Int64)
    @ LinearAlgebra.LAPACK /nobackup/users/vchuravy/builds/julia/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/lapack.jl:38
  [2] gebal!(job::Char, A::Matrix{Float64})
    @ LinearAlgebra.LAPACK /nobackup/users/vchuravy/builds/julia/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/lapack.jl:213
  [3] exp!(A::Matrix{Float64})
    @ LinearAlgebra /nobackup/users/vchuravy/builds/julia/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/dense.jl:590
  [4] exp(A::Matrix{Int64})
    @ LinearAlgebra /nobackup/users/vchuravy/builds/julia/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/dense.jl:555
  [5] _broadcast_getindex_evalf
```

**Comment by vchuravy** [comment]
Or alternativly:

```
julia> exp([1 2; 3 4])

signal (11): Segmentation fault
in expression starting at REPL[1]:1
idamax_k at /nobackup/users/vchuravy/builds/julia/usr/bin/../lib/libopenblas64_.so (unknown line)
idamax_64_ at /nobackup/users/vchuravy/builds/julia/usr/bin/../lib/libopenblas64_.so (unknown line)
dgebal_64_ at /nobackup/users/vchuravy/builds/julia/usr/bin/../lib/libopenblas64_.so (unknown line)
gebal! at /nobackup/users/vchuravy/builds/julia/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/lapack.jl:209
exp! at /nobackup/users/vchuravy/builds/julia/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/dense.jl:590
exp at /nobackup/users/vchuravy/builds/julia/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/dense.jl:555
jl_fptr_args at /nobackup/users/vchuravy/dev/julia/src/gf.c:1993
_jl_invoke at /nobackup/users/vchuravy/dev/julia/src/gf.c:2224
jl_apply_generic at /nobackup/users/vchuravy/dev/julia/src/gf.c:2406
```

**Comment by vchuravy** [comment]
Fixed by #38836
